### PR TITLE
Add indentation to log messages 

### PR DESCRIPTION
**Pull request by Fripe070**
### Checklist
- [X] I have planned and discussed this feature with other contributors.
- [X] I have run basic tests on my code to ensure it is functional.
- [X] I have thoroughly tested my code with various states, contexts and edge cases.
- [X] I have searched for a similar pull request in this repository and didn't find anything relevant.

### Summary
Adds indentation to log messages spanning multiple lines

### Issues
- Implements #236

### CODE REVIEW

**Comment by Fripe070** [comment]
I am unsure of how I want to structure this. The current content of the PR should be fully functional, but is very ugly.
I would love to have some feedback as to the ideal way to tackle this. 
The main challenge lies in the fact that [`discord.utils._ColourFormatter`](https://github.com/Rapptz/discord.py/blob/efe81a67fb55f9f2a67bb84810be729ed8f09bc3/discord/utils.py#L1274-L1315) adds ANSI escape codes

**Comment by m31-galaxy** [comment]
Super naive take here, but surely it wouldn't be that hard to have a function that strips ANSI escapes from a string, and therefore a function which can compute the "visible length" of a string?

**Comment by Fripe070** [comment]
I did that now.
The code is more confusing (algorithm) but more modular

File: breadcord/helpers.py [hunk 1]
```diff
@@ -278,12 +278,29 @@ def __init__(self, to_wrap: logging.Formatter | None = None) -> None:
...
+                    break
+                if all(char in '0123456789;' for char in string[i+2:end]):
+                    i = end + 1
+                    # Avoid IndexError since the escape sequence could be at the end of the string
+                    if i >= len(string):
+                        break
+            new += string[i]
+            i += 1
+        return new
+
```
Comment (Fripe070) [inline breadcord/helpers.py]: Algorithm breakdown:
if we find an "ESC[" we find the next "m" (end of the ANSI code)
if no "m" exists we kill the loop and append the rest of the string, since there can't be any more ANSI codes
if the characters between the "ESC[" and "m" are all numbers or semicolons, we move the index to after the "m"
if the index is at the end of the string, we kill the loop and append the rest of the string

File: breadcord/helpers.py [hunk 2]
```diff
@@ -270,3 +272,29 @@ def make_codeblock(content: str, lang: str | None = None, *, escape_backticks: b
...
     return f'```{lang or ""}\n{content}\n```'
+
+
+class IndentFormatter(logging.Formatter):
+    def __init__(self, to_wrap: logging.Formatter | None = None) -> None:
+        super().__init__()
+        self._wrapped = to_wrap or logging.Formatter()
+
+    def get_prefix_length(self, record: logging.LogRecord) -> int:
+        """Get the length of the prefix of the log message. Will not give wanted results if ANSI is involved."""
```
Comment (m31-galaxy) [inline breadcord/helpers.py:283]: What do you mean by "Will not give wanted results"?

File: breadcord/helpers.py [hunk 3]
```diff
@@ -270,3 +272,29 @@ def make_codeblock(content: str, lang: str | None = None, *, escape_backticks: b
...
+        formatted = self._wrapped.format(logging.LogRecord(
+            name=record.name,
+            level=record.levelno,
+            pathname=record.pathname,
+            lineno=record.lineno,
+            msg='',
+            args=(),
+            exc_info=None,
```
Comment (m31-galaxy) [inline breadcord/helpers.py:291]: Do these not affect the resultant formatting in any way? The default values look a little suspicious to me for potential edge cases such as exception logging, but I'll trust your testing so we can worry about this later if it goes wonky.

**Review by m31-galaxy — APPROVED** [review]

File: breadcord/helpers.py (hunk 2 above)
Comment (Fripe070) [reply breadcord/helpers.py:283]: Forgot to remove that from when ANSI was not supported, that should be changed
